### PR TITLE
add openssh version validation

### LINKED ISSUE #44: Exit on incompatible openssh version
We should check that the openssh install on the system is running > 9.2, and error otherwise. We have seen unexpected behaviours when running with older versions of openssh

### PR DESCRIPTION
**Pull request by jcalisto**
Fixes https://github.com/grafana/pdc-agent/issues/44

The minimum version constraint is documented here:
https://grafana.com/docs/grafana-cloud/connect-externally-hosted/private-data-source-connect/configure-pdc/

### CODE REVIEW

File: pkg/ssh/ssh.go [hunk 1]
```diff
@@ -233,3 +243,43 @@ func extractOptionFromFlag(flag string) (string, string, error) {
...
+func getSSHVersion(ctx context.Context, sshCmd string) (string, error) {
+	out, err := exec.CommandContext(ctx, sshCmd, "-V").CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("failed to run ssh -V command: %w", err)
+	}
+	return string(out), nil
+}
+
+func RequireSSHVersionAbove9_2(version string) error {
+	re := regexp.MustCompile(`OpenSSH_(\d+)\.(\d+)`)
```
Comment (PoorlyDefinedBehaviour) [inline pkg/ssh/ssh.go:266]: Not that it matters here, but we could compile the regex only once

File: pkg/ssh/ssh.go [hunk 2]
```diff
@@ -233,3 +243,43 @@ func extractOptionFromFlag(flag string) (string, string, error) {
...
+
+func getSSHVersion(ctx context.Context, sshCmd string) (string, error) {
+	out, err := exec.CommandContext(ctx, sshCmd, "-V").CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("failed to run ssh -V command: %w", err)
+	}
+	return string(out), nil
+}
+
+func RequireSSHVersionAbove9_2(version string) error {
```
Comment (PoorlyDefinedBehaviour) [inline pkg/ssh/ssh.go:265]: If you want to, this seems like a good candidate for property based testing (generating bunch of random inputs in a specific format)

I usually find bugs doing this

**Review by PoorlyDefinedBehaviour — APPROVED** [review]

File: pkg/ssh/ssh.go [hunk 3]
```diff
@@ -97,6 +101,12 @@ func NewClient(cfg *Config, logger log.Logger, km *KeyManager) *Client {
 func (s *Client) starting(ctx context.Context) error {
 	level.Info(s.logger).Log("msg", "starting ssh client")
 
+	if !s.cfg.SkipSSHValidation {
+		if err := validateSSHVersion(ctx, s.SSHCmd); err != nil {
+			return fmt.Errorf("failed to validate ssh version: %w", err)
```
Comment (dafydd-t) [inline pkg/ssh/ssh.go:106]: I interpret this message as "the validation check failed to complete", which is true sometimes. I think we should distinguish between the two cases (failed to complete validation; warn and continue && validation completed and failed; error)

File: pkg/ssh/ssh.go (hunk 2 above)
Comment (jcalisto) [reply pkg/ssh/ssh.go]: Cool, I was reading about it. In this case, the edge cases are very strict around 9.2 which is fairly simple to test anyway.

File: pkg/ssh/ssh.go (hunk 3 above)
Comment (jcalisto) [reply pkg/ssh/ssh.go]: Sounds reasonable. I refactored the code a bit to follow this, warn if unable to properly fetch the version and error if the version is deemed invalid.
